File: model/src/main/kotlin/net/dungeonhub/model/warning/WarningEvidenceCreationModel.java
package net.dungeonhub.model.warning;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.taubsie.dungeonhub.common.DungeonHubService;
import me.taubsie.dungeonhub.common.entity.model.CreationModel;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class WarningEvidenceCreationModel implements CreationModel {
    private String evidence;
    private long submitter;

    public String toJson() {
        return DungeonHubService.getInstance().getGson().toJson(this);
    }
}

File: model/src/main/kotlin/net/dungeonhub/model/warning/WarningEvidenceCreationModel.kt
package net.dungeonhub.model.warning

import net.dungeonhub.service.MoshiService
import net.dungeonhub.structure.model.CreationModel

class WarningEvidenceCreationModel(
    var evidence: String,
    var submitter: Long
) : CreationModel {
    fun toJson(): String {
        return MoshiService.moshi.adapter(WarningEvidenceCreationModel::class.java).toJson(this)
    }
}